cirq/circuits/circuit.py

```diff
@@ -211,7 +213,9 @@ def __getitem__(self, key: int) -> 'cirq.Moment':
 
     def __getitem__(self, key):
         if isinstance(key, slice):
-            return Circuit(self._moments[key], device=self.device)
+            sliced_circuit = Circuit(device=self.device)
+            sliced_circuit._moments = self._moments[key]
```

Comment:
I'm confused because this looks like List = Moment?

Reply:
See above.  It's a slice, not an element.